apps/whale-api/src/module.api/poolpair.fees.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@nestjs/common'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+import BigNumber from 'bignumber.js'
+import { EstimatedDexFeesInPct } from '@defichain/whale-api-client/dist/api/poolpairs'
+import { PoolPairInfo } from '@defichain/jellyfish-api-core/dist/category/poolpair'
+
+@Injectable()
+export class PoolPairFeesService {
+  constructor (
+    protected readonly rpcClient: JsonRpcClient
+  ) {
+  }
+
+  public async getDexFeesPct (poolPair: PoolPairInfo, fromToken: string, toToken: string): Promise<EstimatedDexFeesInPct | undefined> {
+    const { dexFeeInPctTokenA, dexFeeOutPctTokenA, dexFeeInPctTokenB, dexFeeOutPctTokenB } = poolPair
+    const tokenADirection = poolPair.idTokenA === fromToken ? 'in' : 'out'
+    const tokenBDirection = poolPair.idTokenB === toToken ? 'out' : 'in'
+
+    if (dexFeeInPctTokenA === undefined && dexFeeOutPctTokenA === undefined && dexFeeInPctTokenB === undefined && dexFeeOutPctTokenB === undefined) {
+      return undefined
+    }
```

Comment:
hmm. might have a problem here. asymmetric fee keys are not always set. specially in the old way, only the `token_a_fee_pct`,  `token_b_fee_pct` is set and it's considered as direction `both`. But this will not be updated in `token_a_fee_direction` or `token_b_fee_direction`. you can test this by tweaking this simple test -> https://github.com/JellyfishSDK/jellyfish/blob/a5afb411c106a6cf57bf4676bed0f849b896f6a9/packages/jellyfish-api-core/__tests__/category/masternode/setGov.test.ts#L352

Reply:
Thanks for the explanation, @surangap! Tested this scenario and able to validate that there's no changes needed in this line because `dexFeeInPct**` is populated if direction is `not set`

Regardless if there's no change needed here, I still added [e2e tests ](https://github.com/JellyfishSDK/jellyfish/pull/1601/commits/db69caf13d74c127498d9e663cac9f6b36911eee#diff-d0967f195ebd61af3cfd8c80b467983958ae965caa4857a2d283b4095b6cb07eR498)wherein direction is `not set` (considered as `both`)